app/controllers/devise_token_auth/omniauth_callbacks_controller.rb

```diff
@@ -209,6 +209,10 @@ def render_data_or_redirect(message, data, user_data = {})
 
         # build and redirect to destination url
         redirect_to DeviseTokenAuth::Url.generate(auth_origin_url, data.merge(blank: true))
+
+      elsif using_access_token_strategy?
+        render json: { success: true, data: user_data.merge(data).merge({message: message}) }
```

Comment:
Little suggestion : Extract this into a dedicated method, so it will be really easy to customize the response format.
Have a look at https://github.com/lynndylanhurley/devise_token_auth/blob/master/app/controllers/devise_token_auth/registrations_controller.rb#L65 for an example.